File: apps/api/test_google.py
#!/usr/bin/env python3
import requests
import json
import sys

def test_google_search():
    url = 'http://localhost:3002/v0/scrape'
    
    # 测试 Google 搜索
    search_url = 'https://www.google.com/search?q=2025年+AI+agent'
    data = {'url': search_url}
    
    print(f"正在测试抓取: {search_url}")
    print("-" * 50)
    
    try:
        response = requests.post(url, json=data, timeout=45)
        print(f'HTTP 状态码: {response.status_code}')
        
        if response.status_code == 200:
            result = response.json()
            if result.get('success'):
                content = result['data']['content']
                metadata = result['data']['metadata']
                
                print(f'✅ 抓取成功!')
                print(f'页面标题: {metadata.get("title", "无标题")}')
                print(f'内容长度: {len(content)} 字符')
                print(f'内容预览:')
                print("-" * 30)
                print(content[:800] + "..." if len(content) > 800 else content)
                
                return True
            else:
                print(f'❌ 抓取失败: {result}')
                return False
        else:
            print(f'❌ HTTP错误 {response.status_code}: {response.text[:200]}')
            return False
            
    except requests.exceptions.Timeout:
        print('❌ 请求超时')
        return False
    except Exception as e:
        print(f'❌ 请求异常: {e}')
        return False

if __name__ == "__main__":
    success = test_google_search()
    sys.exit(0 if success else 1)

File: apps/api/test_google.js
const https = require('https');
const http = require('http');

function makeRequest(url, data) {
    return new Promise((resolve, reject) => {
        const postData = JSON.stringify(data);
        
        const options = {
            hostname: 'localhost',
            port: 3002,
            path: '/v0/scrape',
            method: 'POST',
            headers: {
                'Content-Type': 'application/json',
                'Content-Length': Buffer.byteLength(postData)
            }
        };

        const req = http.request(options, (res) => {
            let responseData = '';
            
            res.on('data', (chunk) => {
                responseData += chunk;
            });
            
            res.on('end', () => {
                try {
                    const result = JSON.parse(responseData);
                    resolve({ statusCode: res.statusCode, data: result });
                } catch (e) {
                    resolve({ statusCode: res.statusCode, data: responseData });
                }
            });
        });

        req.on('error', (err) => {
            reject(err);
        });

        req.setTimeout(45000, () => {
            req.destroy();
            reject(new Error('请求超时'));
        });

        req.write(postData);
        req.end();
    });
}

async function testGoogleSearch() {
    const searchUrl = 'https://www.google.com/search?q=2025年+AI+agent';
    
    console.log(`正在测试抓取: ${searchUrl}`);
    console.log('-'.repeat(50));
    
    try {
        const response = await makeRequest('http://localhost:3002/v0/scrape', {
            url: searchUrl
        });
        
        console.log(`HTTP 状态码: ${response.statusCode}`);
        
        if (response.statusCode === 200 && response.data.success) {
            const { content, metadata } = response.data.data;
            
            console.log('✅ 抓取成功!');
            console.log(`页面标题: ${metadata.title || '无标题'}`);
            console.log(`内容长度: ${content.length} 字符`);
            console.log('内容预览:');
            console.log('-'.repeat(30));
            console.log(content.substring(0, 800) + (content.length > 800 ? '...' : ''));
            
            return true;
        } else {
            console.log('❌ 抓取失败:', response.data);
            return false;
        }
        
    } catch (error) {
        console.log('❌ 请求异常:', error.message);
        return false;
    }
}

testGoogleSearch().then(success => {
    process.exit(success ? 0 : 1);
});
